### PR TITLE
Correct argv between integratord and integration scripts

### PR DESCRIPTION
**Pull request by nicolaspapp**
|Related issue|
|---|
| #4798 |

## Description
Solve mismatch in integration scripts when debug flag is set to active 

## Configuration options
Establish configuration for the 3 integrations:

```
<integration>
  <name>slack</name>
  <hook_url>https://hooks.slack.com/services/...</hook_url> <!-- Replace with your Slack hook URL -->
  <alert_format>json</alert_format>
</integration>

<integration>
  <name>pagerduty</name>
  <api_key>API_KEY</api_key> <!-- Replace with your PagerDuty API key -->
</integration>

<integration>
  <name>virustotal</name>
  <api_key>API_KEY</api_key> <!-- Replace with your VirusTotal API key -->
  <group>syscheck</group>
  <alert_format>json</alert_format>
</integration>
```

## Tests
Manually tested the 3 existen integrations

### CODE REVIEW

**Comment by wazuhci** [comment]
The stress test has been launched. The charts will be posted when they are ready.
I will use the following configuration:
### General
- Commit: e24c127
- Packages: 
  - [manager 3.12.1 0.commite24c127 rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-manager-3.12.1-0.commite24c127.x86_64.rpm)
  - [agent 3.12.1 0.commite24c127 rpm](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/rpm/var/wazuh-agent-3.12.1-0.commite24c127.x86_64.rpm)
  - [manager 3.12.1 0.commite24c127 deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-manager_3.12.1-0.commite24c127_amd64.deb)
  - [agent 3.12.1 0.commite24c127 deb](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/deb/var/wazuh-agent_3.12.1-0.commite24c127_amd64.deb)
  - [agent 3.12.1 0.commite24c127 msi](https://packages-dev.wazuh.com/warehouse/pullrequests/3.12/windows/wazuh-agent-3.12.1-0.commite24c127.msi)

### Stress test parameters:
- Manager system: `centos`
- Launch CentOS agent: `true`
- Launch Ubuntu agent: `true`
- Launch Windows agent: `true`
- Test duration: `60`
- Scan frequency: `10`
- Modules to test: 
  - logcollector: `true`
  - syscheck: `true`
  - rootcheck: `true`
  - sca: `true`
  - osquery: `true`
  - syscollector: `true`
  - active-response: `true`
  - azure-logs: `true`
  - cis-cat: `true`
  - docker-listener: `true`
  - open-scap: `true`
  - vulnerability-detector: `false`
- FIM special test: `None`
- FIM sleep: `300`
- FIM max files: `25000`
- SCA sleep: `30`
- SCA max registers: `5000`
- Docker image name: `hello-world`
- Docker sleep: `5`
- Docker jobs: `10`
- Logcollector special test: `None`
- Logcollector max files: `50000`
- Logcollector creation sleep: `300`
- Logcollector write files: `1000`
- Logcollector write sleep: `10`
- Vuln sleep: `60`
- Authd port: `1515`
- Syscollector package: `imagemagick.app`
- Syscollector package version: `7.0.8.48`

Regards!.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** e24c127b26844b9b8de8ab9d190b4a731bffea7f
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B869/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B869/ossec.log)
**Coverage:** 

```json
Lines executed:31.40% of 1860

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B869/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.12.1-B869.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B869/Analysis_dynamic_valgrind-v3.12.1-B869.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** e24c127b26844b9b8de8ab9d190b4a731bffea7f
**Parameters:** track-origins=yes leak-check=full keep-stacktraces=alloc-and-free track-fds=yes num-callers=20 show-leak-kinds=definite,indirect
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B870/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B870/ossec.log)
**Coverage:** 

```json
Lines executed:26.47% of 2996

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B870/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.12.1-B870.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.12.1/B870/Analysis_dynamic_valgrind-v3.12.1-B870.log)

File: src/os_integrator/integrator.c [hunk 1]
```diff
@@ -383,7 +383,7 @@ void OS_IntegratorD(IntegratorConfig **integrator_config)
             if(temp_file_created == 1)
             {
                 int dbg_lvl = isDebug();
-                snprintf(exec_full_cmd, 4095, "%s %s %s %s %s", integrator_config[s]->path, exec_tmp_file, integrator_config[s]->apikey == NULL?"":integrator_config[s]->apikey, integrator_config[s]->hookurl==NULL?"":integrator_config[s]->hookurl, dbg_lvl <= 0 ? "" : "debug");
+                snprintf(exec_full_cmd, 4095, "%s %s %s %s %s", integrator_config[s]->path, exec_tmp_file, integrator_config[s]->apikey == NULL ? "None" : integrator_config[s]->apikey, integrator_config[s]->hookurl == NULL ? "None" : integrator_config[s]->hookurl, dbg_lvl <= 0 ? "" : "debug");
```
Comment (vikman90) [inline src/os_integrator/integrator.c:386]: I'm not sure Integratord should replace the empty strings with "None".

**Pull request merged**